vscode/vscode/vscode.ml

```diff
@@ -1810,39 +1812,275 @@ module Workspace = struct
     [@@js.global "vscode.workspace.openTextDocument"]
 end
 
+module TreeItemCollapsibleState = struct
+  type t =
+    | None [@js 0]
+    | Collapsed [@js 1]
+    | Expanded [@js 2]
+  [@@js.enum] [@@js]
+end
+
+module TreeItemLabel = struct
+  type t = private (* interface *) Ojs.t [@@js]
+
+  val label : t -> string [@@js.get]
+
+  val highlights : t -> (int * int) list or_undefined [@@js.get]
+
+  val create : label:string -> ?highlights:(int * int) list -> unit -> t
+    [@@js.builder]
+end
+
+module ThemeIcon = struct
+  type t = private (* class *) Ojs.t [@@js]
+
+  val make : id:string -> ?color:ThemeColor.t -> unit -> t
+    [@@js.new "vscode.ThemeIcon"]
+
+  val file : t [@@js.global "vscode.ThemeIcon.File"]
+
+  val folder : t [@@js.global "vscode.ThemeIcon.Folder"]
+
+  val id : t -> string [@@js.get]
+
+  val color : t -> ThemeColor.t option [@@js.get]
+end
+
+module TreeItem = struct
+  type t = private (* class *) Ojs.t [@@js]
+
+  module LightDarkIcon = struct
+    type t =
+      { light : ([ `String of string | `Uri of Uri.t ][@js.union])
+      ; dark : ([ `String of string | `Uri of Uri.t ][@js.union])
+      }
+    [@@js]
+
+    let t_of_js js_val =
+      let light_js = Ojs.get js_val "light" in
+      let dark_js = Ojs.get js_val "dark" in
+      let light =
+        if Ojs.has_property light_js "parse" then
+          `Uri (Uri.t_of_js light_js)
+        else
+          `String ([%js.to: string] light_js)
+      in
+      let dark =
+        if Ojs.has_property dark_js "parse" then
+          `Uri (Uri.t_of_js dark_js)
```

Comment:
`Uri.t_of_js` can be `[%js.to: Uri.t]` here.

Reply:
Are there any advantages of using the PPX, or is it just a convention?

Reply:
I think the PPX just calls the function, but using it here keeps it consistent with the rest of the bindings and makes it clear which type the JS value is being converted to.